### PR TITLE
Solution by DK

### PR DESCRIPTION
**Pull request by DimaK123**
[DEMO LINK](https://Dimak123.github.io/react_tabs/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -10,33 +12,29 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTabId] = useState('tab-1');
+
+  const currentTab = tabs.find(tab => tab.id === selectedTabId) || tabs[0];
+
+  const onTabSelected = (tab: Tab) => {
```
Comment (polosanya) [inline src/App.tsx:19]:
```suggestion
  const handleTabSelected = (tab: Tab) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

File: src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -1 +1,47 @@
...
+        <div className="tabs is-boxed">
+          <ul>
+            {tabs.map(tab => {
+              const { id, title } = tab;
+
+              return (
+                <li
+                  key={id}
+                  data-cy="Tab"
+                  className={classNames({ 'is-active': selectedTabId === id })}
```
Comment (polosanya) [inline src/components/Tabs/Tabs.tsx:30]: Compare it with the currentTab.id

File: src/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -1 +1,47 @@
...
+                >
+                  <a href={`#${id}`} data-cy="TabLink" onClick={() => currentTab !== tab && onTabSelected(tab)}>
```
Comment (polosanya) [inline src/components/Tabs/Tabs.tsx:32]:
```suggestion
                  <a 
                    href={`#${id}`} 
                    data-cy="TabLink" 
                    onClick={() => currentTab !== tab && onTabSelected(tab)}
                  >
```
Also, don't compare objects like that
image

File: src/App.tsx [hunk 4]
```diff
@@ -10,33 +12,29 @@ export const tabs = [
...
+    if (tabs.find(someTab => someTab.id === tab.id)) {
+      setSelectedTabId(tab.id);
+    } else {
+      setSelectedTabId(tabs[0].id);
```
Comment (polosanya) [inline src/App.tsx:23]: You already check it when finding a tab, so there is actually no need to double check

File: src/components/Tabs/Tabs.tsx [hunk 5]
```diff
@@ -1 +1,47 @@
...
+  onTabSelected: (tab: Tab) => void,
+};
+
+export const Tabs:React.FC<Props> = React.memo(
+  ({
+    tabs,
+    selectedTabId,
+    onTabSelected,
+  }) => {
+    const currentTab = tabs.find(tab => tab.id === selectedTabId) || tabs[0];
```
Comment (polosanya) [inline src/components/Tabs/Tabs.tsx:17]: You do the same here and in App. 
You can create a helper function in order not to duplicate yourself

**Review by polosanya — CHANGES_REQUESTED** [review]
Let's improve your code a bit)

File: src/components/Tabs/Tabs.tsx (hunk 5 above)
Comment (DimaK123) [reply src/components/Tabs/Tabs.tsx:17]: Thanks for the advice. I hope that I've done everything correctly, but unfortunately, the test still failed. I see the error message in the terminal.
Why onTabFinder is not a function ?


изображение

**Review by TarasHoliuk — APPROVED** [review]
LGTM :+1:

**Review by agniya-i — APPROVED** [review]

**Pull request closed without merging**